api/include/opentelemetry/trace/span.h

```diff
@@ -36,11 +37,16 @@ struct StartSpanOptions
   core::SystemTimestamp start_system_time;
   core::SteadyTimestamp start_steady_time;
 
+  // Optionally set attributes at Span creation from the given key/value pairs.
+  //
+  // Attributes will be processed in order, previous attributes with the same
+  // key will be overwritten.
+  nostd::span<std::pair<nostd::string_view, common::AttributeValue>> attributes;
```

Comment:
nostd::span is a non-owning reference. (as is string_view). There are non-trivial lifetime issues putting this into StartSpanOptions. I'm not sure what to do here.

Reply:
I'll think more about his.

One solution would be to make this an argument to `StartSpan` (in addition to options). Then there'd be no lifetime issues of storing this in a struct.

Reply:
I went with the attributes as an additional argument to `StartSpan`. This should fix any lifetime issues.